nautobot/extras/api/serializers.py

```diff
@@ -569,6 +570,30 @@ class Meta:
         model = JobQueue
         fields = "__all__"
 
+    def validate(self, data):
+        if self.instance:
+            secrets_group = data.get("secrets_group", self.instance.secrets_group)
+            context = data.get("context", self.instance.context)
+            queue_type = data.get("queue_type", self.instance.queue_type)
+
+            errors = {}
+            if secrets_group and queue_type == JobQueueTypeChoices.TYPE_CELERY:
+                error_message = (
+                    f"Secrets groups are not allowed on job queues of type {JobQueueTypeChoices.TYPE_CELERY}"
+                )
+
+                if "secrets_group" in data:
+                    errors["secrets_group"] = [error_message]
+
+            if context and queue_type == JobQueueTypeChoices.TYPE_CELERY:
+                error_message = f"Context is not allowed on job queues of type {JobQueueTypeChoices.TYPE_CELERY}"
+                if "context" in data:
+                    errors["context"] = [error_message]
+
+                raise serializers.ValidationError(errors)
```

Comment:
This feels off to me - so if the instance has a secrets group or context, but `secrets_group` or `context` isn't in `data`, we will raise an empty ValidationError?

Also the indentation on that `raise` seems off to me.

Reply:
Yeah it should raise the errors regardless. Thanks for catching it.